src/app/core/store/selectors/addresses.selectors.ts

```diff
@@ -39,7 +42,7 @@ export const getAllAddresses = createSelector(
 );
 export const getAllAddressesForAssetId = (aAssetId: number) => createSelector(
   getAllAddresses,
-  (aAddresses) => aAddresses.filter(aAddress => aAddress.getAssetId() === +aAssetId)
+  (aAddresses) => aAddresses.filter(aAddress => aAddress).filter(aAddress => aAddress.getAssetId() === +aAssetId)
```

Comment:
If we need to watch out for empty addresses, why not just `(aAddresses) => aAddresses.filter(aAddress => aAddress && aAddress.getAssetId() === +aAssetId)` instead of filtering twice?